MailCore/Cache/MailboxManager/MailboxManager+Thread.swift

```diff
@@ -591,6 +604,21 @@ public extension MailboxManager {
         writableRealm.delete(orphanMessages)
     }
 
+    private func resetFolder(_ folder: Folder) async throws {
+        try writeTransaction { realm in
+            guard let liveFolder = folder.fresh(using: realm) else { return }
+
+            liveFolder.remainingOldMessagesToFetch = Constants.messageQuantityLimit
+            liveFolder.oldMessagesUidsToFetch.removeAll()
```

Comment:
I didn't use Folder.resetHistoryInfo because I want this block to be explicit with the properties it resets.